library/common/types/c_types.h

```diff
@@ -50,7 +50,7 @@ void envoy_noop_release(void* context);
  * Holds raw binary data as an array of bytes.
  */
 typedef struct {
-  size_t length;
+  int length;
```

Comment:
I'd make this `size_t`, but I'd store the actual length - when we see that `bytes` is nullptr, but `length` is non-zero, we know an error occurred.

Reply:
We can put a comment on the utility functions' docstrings clarifying this.